src/main/java/ch/njol/skript/config/Node.java

```diff
@@ -115,36 +116,48 @@ public void move(final SectionNode newParent) {
 		newParent.add(this);
 	}
 	
-	@SuppressWarnings("null")
-	private final static Pattern linePattern = Pattern.compile("^((?:[^#]|##)*)(\\s*#(?!#).*)$");
-	
 	/**
 	 * Splits a line into value and comment.
 	 * <p>
-	 * Whitespace is preserved (whitespace in front of the comment is added to the value), and any ## in the value are replaced by a single #. The comment is returned with a
+	 * Whitespace is preserved (whitespace in front of the comment is added to the value). The comment is returned with a
 	 * leading #, except if there is no comment in which case it will be the empty string.
 	 * 
 	 * @param line
 	 * @return A pair (value, comment).
 	 */
-	public static NonNullPair<String, String> splitLine(final String line) {
+	public static NonNullPair<String, String> splitLine(String line) {
 		if (line.trim().startsWith("#"))
 			return new NonNullPair<>("", line.substring(line.indexOf('#')));
-		final Matcher m = linePattern.matcher(line);
-		boolean matches = false;
-		try {
-			matches = line.contains("#") && m.matches();
-		} catch (StackOverflowError e) { // Probably a very long line
-			handleNodeStackOverflow(e, line);
+
+		// idea: find first # that is not within a string. Use " and % to determine string state.
+		int length = line.length();
+		boolean inString = false;
+		// find next " or %
+		for(int i = 0; i < length; i++) {
+			char c = line.charAt(i);
+			// check for " and % which change the inString state
+			if (c == '%' || c == '"') {
+				// skip if doubled
+				if (i + 1 < length && line.charAt(i+1) == c) {
+					i++;
+					continue;
+				}
+				// just % indicates expression start/end
+				// just " indicates quote start/end
+				// (inString -> exiting string and entering expression and vice versa)
+				inString = !inString;
+			// looking for # outside of strings
+			} else if (c == '#' && !inString) {
```

Comment:
Should be a little better in performance.
Is it worth it to create a final static field for comment symbol?

This also means variables can no longer use hashtags which should be stated in the release log as it may be considered breaking change
![image](https://github.com/SkriptLang/Skript/assets/20037329/48b5179c-2a86-44cd-b511-f02b0f544298)

```suggestion
			} else if (!inString && c == '#') {
```

Reply:
Oh I'll add an exception for that, we should still allow that

Reply:
We could even add a cached-value check if line has a hashtag then we proceed with comment checks

Reply:
I didn't short-circuit with a contains() because it felt like a needless extra loop, but since we hae to handle variables now too, it's a bit more complicated and might be worth it.